ompi/mca/coll/adapt/coll_adapt_ireduce.c

```diff
@@ -34,109 +34,109 @@ typedef int (*mca_coll_adapt_ireduce_fn_t) (const void *sbuf,
                                             ompi_request_t ** request,
                                             mca_coll_base_module_t * module, int ireduce_tag);
 
-static mca_coll_adapt_algorithm_index_t mca_coll_adapt_ireduce_algorithm_index[] = {
-    {0, (uintptr_t)mca_coll_adapt_ireduce_tuned},
-    {1, (uintptr_t) mca_coll_adapt_ireduce_binomial},
-    {2, (uintptr_t) mca_coll_adapt_ireduce_in_order_binomial},
-    {3, (uintptr_t) mca_coll_adapt_ireduce_binary},
-    {4, (uintptr_t) mca_coll_adapt_ireduce_pipeline},
-    {5, (uintptr_t) mca_coll_adapt_ireduce_chain},
-    {6, (uintptr_t) mca_coll_adapt_ireduce_linear},
+static ompi_coll_adapt_algorithm_index_t ompi_coll_adapt_ireduce_algorithm_index[] = {
+    {0, (uintptr_t)ompi_coll_adapt_ireduce_tuned},
+    {1, (uintptr_t) ompi_coll_adapt_ireduce_binomial},
+    {2, (uintptr_t) ompi_coll_adapt_ireduce_in_order_binomial},
+    {3, (uintptr_t) ompi_coll_adapt_ireduce_binary},
+    {4, (uintptr_t) ompi_coll_adapt_ireduce_pipeline},
+    {5, (uintptr_t) ompi_coll_adapt_ireduce_chain},
+    {6, (uintptr_t) ompi_coll_adapt_ireduce_linear},
```

Comment:
A type is defined for every collective function in ompi/mca/coll/coll.h:
`coll.h

    typedef int (*mca_coll_base_module_ireduce_fn_t)
      (const void *sbuf, void* rbuf, int count, struct ompi_datatype_t *dtype,
       struct ompi_op_t *op, int root, struct ompi_communicator_t *comm, ompi_request_t ** request,
       struct mca_coll_base_module_2_3_0_t *module);
`
Why using uintptr_t instead ?

Reply:
nvm, ompi_coll_adapt_algorithm_index_t uses a generic pointer